sh/reset_repos.sh

```diff
@@ -21,14 +22,14 @@ if [ ! -f "$URL_FILE" ]; then
 fi
 
 while IFS= read -r FORK_URL; do
-  UPSTREAM_URL=$(echo "$FORK_URL" | sed 's/opentofu/terraform-providers/')
+  UPSTREAM_URL=${FORK_URL//opentofu/terraform-providers}
 
   echo "Got fork URL: $FORK_URL"
   echo "Got upstream URL: $UPSTREAM_URL"
 
   git clone "$FORK_URL"
   REPO_NAME="$(basename "$FORK_URL" .git)"
-  cd "$REPO_NAME"
+  popd "$REPO_NAME" || exit 1
```

Comment:
What's the advantage of this approach?

pushd/popd are rather less known than cd, so if there is no big advantage I'd prefer to keep this as-is.

Reply:
@cube2222  In the context we're discussing, using `pushd` and `popd` doesn't bring much to the table. However, keeping track of directories (`dirs`) generally works better than constantly switching between contexts, especially when you need to return to where you started.
To make the script more foolproof, I suggested to add `|| exit 1` in case `cd` fails. This ensures that if something goes wrong, the script doesn't leave a mess behind. Another option would be to include set -o errexit`, which makes the script exit right away if any part of it fails.
WDYT ? :-)

Reply:
> Another option would be to include set -o errexit`

I like this proposal much better!

Reply:
I reverted the changes and added `set -o errexit` in this commit: [1cc708c](https://github.com/opentofu/scripts/pull/4/commits/1cc708c17980968a42a93449d5257606c0e60569)